common/shared.c

```diff
@@ -1464,7 +1464,7 @@ size_t datatype_to_size(enum fi_datatype datatype)
 	case FI_LONG_DOUBLE_COMPLEX:
 		return sizeof(OFI_COMPLEX(long_double));;
 	default:
-		return 0;
+		assert(0);
```

Comment:
We an assert() here, but we should still return a value in the non-debug build or we'll end up with a code path that doesn't return anything.

Reply:
`assert(0)` is pretty anti-social for middleware (it will dump a corefile).

Reply:
This is fabtests, not libfabric.  We could probably convert this function into an array access as well.

Reply:
Even so -- this is a diagnostic test that sysadmins run, right?   Dumping core on them would be fairly anti-social.  Corefiles are only useful for developers, IMHO.

Reply:
The only way we get here is if we coded this wrong and passed in a non-enum value into the function, ignoring the compiler warnings about that along the way.  And the assert() only shows up in a debug build anyway. And why is a sys admin running a debug version of fabtests if he's not wanting to develop with it.  And if we can't add asserts into code for our own development purposes, then we might as well remove the ability to build debug code at all.  And I'm done ranting now.